packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/migrate-idp-resources.ts

```diff
@@ -0,0 +1,41 @@
+import { $TSObject, JSONUtilities, pathManager, Template } from '@aws-amplify/amplify-cli-core';
+
+const { readJson } = JSONUtilities;
+const { getCurrentCfnTemplatePathFromBuild, getCurrentCloudRootStackCfnTemplatePath } = pathManager;
+
+export const migrateResourcesToCfn = (resourceName: string): boolean => {
+  const authCfnTemplatePath = getCurrentCfnTemplatePathFromBuild('auth', resourceName);
+  const authCfnTemplate: Template | undefined = readJson(authCfnTemplatePath, { throwIfNotExist: false });
+  const lambdaCalloutCreatedInCloud = authCfnTemplate?.Resources?.HostedUIProvidersCustomResource?.Type === 'AWS::Lambda::Function';
+  const providerCreatedInCloud = hasHostedProviderResources(authCfnTemplate);
+
+  return lambdaCalloutCreatedInCloud && !providerCreatedInCloud;
+};
+
+const hasHostedProviderResources = (authCfnTemplate: Template | undefined): boolean => {
+  return (
+    authCfnTemplate?.Resources?.HostedUIFacebookProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider' ||
+    authCfnTemplate?.Resources?.HostedUIGoogleProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider' ||
+    authCfnTemplate?.Resources?.HostedUILoginWithAmazonProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider' ||
+    authCfnTemplate?.Resources?.HostedUISignInWithAppleProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider'
+  );
+};
+
+export const exportHostedUIProvidersFromCurrCloudRootStack = (resourceName: string, updatedUIProviderCreds: $TSObject[]): $TSObject[] => {
+  const rootCfnTemplatePath = getCurrentCloudRootStackCfnTemplatePath();
+  const rootCfnTemplate: Template | undefined = readJson(rootCfnTemplatePath, { throwIfNotExist: false });
+  const resource = (rootCfnTemplate?.Resources || {})[`auth${resourceName}`];
+  const currProviderCreds = JSON.parse(resource?.Properties?.Parameters?.hostedUIProviderCreds || '[]');
```

Comment:
I could be mistaken, but I think these credentials are only in the root stack on the push where auth is added. Subsequent pushes don't have the credentials in the stack.
If we need the credentials after the initial auth deploy, we have to make an SDK call to Cognito to fetch them.

Or I could be wrong and these are available in the root stack on any push

Reply:
If I do an `amplify pull`, the credentials are pulled. Since a pull happens after a push, I was pretty sure they were supposed to be there... I can ask and see if that is the expected behavior because if it's not, then we should make the SDK calls instead.

Reply:
I can double check later this afternoon, but the scenario I'm worried about is:
1. add and push auth with an oauth provider
2. do some other unrelated deployment
3. try to do this migration

I _think_ after step 2 the credentials are no longer in the root stack, even after pulling

Reply:
Just tried it out and confirmed the behavior I thought.
1. `amplify init`
2. `amplify add auth` // configure one or more 3P oauth providers
3. `amplify push`
4. observe that the hostedUIProviderCreds are in the root stack in #ccb build dir
5. `amplify add function` // select defaults
6. `amplify push`
7. observe that hostedUIProviderCreds is now an empty array in the root stack in #ccb build dir

Reply:
Yep, you're right. I'll need to change this